### DISCUSSION TITLE
Add any true constraint

### DISCUSSION THREAD

**Discussion by ronenbh1** [google/or-tools#2835] [Routing (and legacy CP) questions]
Hello,
I'm trying to add a constraint that validates that at least one cumul var of end index of multiple dimentions is equal to the cumul var of end index of a counter dimention e.g.:
```python
for vehicle_index in vehicles:
    end_index = routing.End(vehicle_index)
    cnt_checks = [counter_dimension.CumulVar(end_index) == dimension.CumulVar(end_index) for dimension in dimensions]
    var_checks = [check.Var() for check in cnt_checks]
    routing.solver().Add(routing.solver().SumGreaterOrEqual(var_checks , 1))
```
But it doesn't work, any suggestions?
I'm using v.9.1.9490
Thanks,
Ronen

**Comment by Mizux** [comment]
did you try:
```python
solver = routing.solver()
for vehicle_index in vehicles:
    end_index = routing.End(vehicle_index)
    cnt_checks = [counter_dimension.CumulVar(end_index) == dimension.CumulVar(end_index) for dimension in dimensions]
    var_checks = [check.Var() for check in cnt_checks]
    solver.Add(solver.Sum(var_checks) >= 1)
```

**Reply by ronenbh1** [reply]
Thanks for your quick reply.
I have tried your suggestion but it didn't solve it.
Maybe there is another way to write it?
In a "pythonic way" I would have wrote it like:
`solver.Add(any([counter_dimension.CumulVar(end_index) == dimension.CumulVar(end_index) for dimension in dimensions]))`

**Reply by Mizux** [reply]
or maybe it is too complicate to find a solution where **all** vehicle have at least one dimension cumulVar end's node equal to your counter dimension...

**Reply by ronenbh1** [reply]
Ok my mistake... I missed a slack being used.
SumGreaterOrEqual works fine.
Thanks a lot @Mizux for being so supportive every time!

**Reply by meitalissers** [reply]
> or maybe it is too complicate to find a solution where **all** vehicle have at least one dimension cumulVar end's node equal to your counter dimension...

Hello, 

Following the question above, is there a way to extract the Min value from the CumulVar? At this point I still do not have the solution.
I need to compare the values ​​of the Cumul variables without regard to slack.

Thank you in advance
Meital
